src/index.tsx

```diff
@@ -826,6 +826,98 @@ export function useBandwidthEstimation() {
+    const eventListener = eventEmitter.addListener(
+      'StatisticsUpdated',
+      (stats) => {
+        if (statistics.length === MAX_SIZE) {
+          statistics.shift();
+        }
+        statistics.push(processIncomingStats(stats));
+        setStatistics([...statistics]);
+      }
+    );
+    return () => eventListener.remove();
+  }, []);
+
+  /**
+   * Collects current RTC statistics.
+   */
+  const getStatistics = useCallback(async () => {
+    await Membrane.getStatistics();
+  }, []);
+
+  /**
+   * Clears statistics array,
+   * efectivly starts collecting statistics from the beginning.
+   */
+  const clearStatistics = useCallback(() => {
+    setStatistics([]);
+  }, []);
+
+  // Calculates diff between pervious and current stats,
+  // providing end users with a per second metric.
+  const processIncomingStats = useCallback(
+    (stats) => {
+      Object.keys(stats).forEach((obj) => {
+        if (obj.includes('Inbound')) {
+          if (Object.keys(statistics[statistics.length - 1]).includes(obj)) {
+            stats[obj]['packetsLost/s'] =
+              stats[obj]['packetsLost'] -
+              statistics[statistics.length - 1][obj]['packetsLost'];
+            stats[obj]['packetsReceived/s'] =
+              stats[obj]['packetsReceived'] -
+              statistics[statistics.length - 1][obj]['packetsReceived'];
+            stats[obj]['bytesReceived/s'] =
+              stats[obj]['bytesReceived'] -
+              statistics[statistics.length - 1][obj]['bytesReceived'];
+            stats[obj]['framesReceived/s'] =
+              stats[obj]['framesReceived'] -
+              statistics[statistics.length - 1][obj]['framesReceived'];
+            stats[obj]['framesDropped/s'] =
+              stats[obj]['framesDropped'] -
+              statistics[statistics.length - 1][obj]['framesDropped'];
+          } else {
+            stats[obj]['packetsLost/s'] = 0;
+            stats[obj]['packetsReceived/s'] = 0;
+            stats[obj]['bytesReceived/s'] = 0;
+            stats[obj]['framesReceived/s'] = 0;
+            stats[obj]['framesDropped/s'] = 0;
+          }
+          return stats;
+        }
+        // Outbound
+        if (Object.keys(statistics[statistics.length - 1]).includes(obj)) {
+          stats[obj]['bytesSent/s'] =
+            stats[obj]['bytesSent'] -
+            statistics[statistics.length - 1][obj]['bytesSent'];
+          stats[obj]['packetsSent/s'] =
+            stats[obj]['packetsSent'] -
+            statistics[statistics.length - 1][obj]['packetsSent'];
+          stats[obj]['framesEncoded/s'] =
+            stats[obj]['framesEncoded'] -
+            statistics[statistics.length - 1][obj]['framesEncoded'];
+        } else {
+          stats[obj]['bytesSent/s'] = 0;
+          stats[obj]['packetsSent/s'] = 0;
+          stats[obj]['framesEncoded/s'] = 0;
+        }
+      });
+      return stats;
+    },
+    [statistics]
+  );
+
+  return { statistics, getStatistics, clearStatistics };
```

Comment:
This API is weird. `getStatistics` suggests that it's returning statistics. I have two propositions:
1.  In `index.tsx` there'd be just one function `getStatistics` that returns the promise that resolves to statistics. You can put the statistics in `promise.resolve(statistics)` on the native side of `getStatistics`. That would be easy, but user would be responsible for calling it every second and processing the stats. 
2. Make a hook `useStatistics` that just returns statistics that are automatically updated but without exposing `getStatistics` to the user. User would just specify the update interval in the hook and the hook takes care of calling `getStatistics` when the hook is mounted and processing the stats. 

Choose which would be better, I'm leaning towards the second option.

Reply:
Done.